samza-core/src/main/java/org/apache/samza/container/grouper/task/TaskAssignmentManager.java

```diff
@@ -21,32 +21,72 @@
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
-import org.apache.samza.coordinator.stream.CoordinatorStreamManager;
-import org.apache.samza.coordinator.stream.messages.CoordinatorStreamMessage;
-import org.apache.samza.coordinator.stream.messages.Delete;
+
+import org.apache.samza.config.Config;
+import org.apache.samza.config.JobConfig;
+import org.apache.samza.container.SamzaContainerContext;
+import org.apache.samza.coordinator.stream.CoordinatorStreamKeySerde;
+import org.apache.samza.coordinator.stream.CoordinatorStreamValueSerde;
 import org.apache.samza.coordinator.stream.messages.SetTaskContainerMapping;
+import org.apache.samza.metadatastore.MetadataStore;
+import org.apache.samza.metadatastore.MetadataStoreFactory;
+import org.apache.samza.metrics.MetricsRegistry;
+import org.apache.samza.serializers.Serde;
+import org.apache.samza.util.Util;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-
 /**
  * Task assignment Manager is used to persist and read the task-to-container
- * assignment information from the coordinator stream
+ * assignment information from the coordinator stream.
  * */
 public class TaskAssignmentManager {
```

Comment:
Can we make "LocalityManager" and "TaskAssignmentManager" have a common base class that encapsulates 
config, serdes, Metadatastore?
This way any additions to Metadatastore can naturally propagate to all "*Managers", and will provide a way to enforce somethings across *Manager classes